Description: Remove -ffile-prefix-map= from CMake module generated compiler flags.
Author: Tobias Frost <tobi@debian.org>
Forwarded: no, Debian specific
Applied-Upstream: <version|URL|commit, identifies patches merged upstream, optional>
Reviewed-by: <name and email of a reviewer, optional>
Last-Update: 2024-05-24 <YYYY-MM-DD, last update of the meta-information, optional>
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1322,6 +1322,9 @@
 endif()
 
 # Configure and install cmake config file
+# remove -ffile-prefix-map=
+string(REGEX REPLACE " -ffile-prefix-map=.*=\\. " " " CMAKE_C_FLAGS_CLEANED ${CMAKE_C_FLAGS})
+string(REGEX REPLACE " -ffile-prefix-map=.*=\\. " " " CMAKE_CXX_FLAGS_CLEANED ${CMAKE_CXX_FLAGS})
 configure_file("${CMAKE_SOURCE_DIR}/adm/templates/OpenCASCADEConfig.cmake.in" "OpenCASCADEConfig.cmake" @ONLY)
 install(FILES "${CMAKE_BINARY_DIR}/OpenCASCADEConfig.cmake" DESTINATION  "${INSTALL_DIR_CMAKE}")
 
--- a/adm/templates/OpenCASCADEConfig.cmake.in
+++ b/adm/templates/OpenCASCADEConfig.cmake.in
@@ -43,8 +43,8 @@
 set (OpenCASCADE_RESOURCE_DIR "@CMAKE_INSTALL_FULL_DATAROOTDIR@/opencascade/resources")
 
 # The C and C++ flags added by OpenCASCADE to the cmake-configured flags.
-set (OpenCASCADE_C_FLAGS      "@CMAKE_C_FLAGS@")
-set (OpenCASCADE_CXX_FLAGS    "@CMAKE_CXX_FLAGS@")
+set (OpenCASCADE_C_FLAGS      "@CMAKE_C_FLAGS_CLEANED@")
+set (OpenCASCADE_CXX_FLAGS    "@CMAKE_CXX_FLAGS_CLEANED@")
 @SET_OpenCASCADE_LINKER_FLAGS@
 
 # List of available OpenCASCADE modules.
